### PR TITLE
[Affinda] New Components

### LINKED ISSUE #7815: [Components] Affinda
# Affinda
## URLs
- https://docs.affinda.com/reference/getallorganizations-1
- https://docs.affinda.com/reference/getallcollections
- https://docs.affinda.com/reference/getallworkspaces-1
### document-parse-completed
Emit an event when a document parsing is completed
- https://docs.affinda.com/docs/webhooks
- https://docs.affinda.com/reference/getalldocuments
### document-validated
Emit an event when a document is validated
- https://docs.affinda.com/docs/webhooks
- https://docs.affinda.com/reference/getalldocuments
### upload-document
Create an action for uploading a document to Affinda
- https://docs.affinda.com/reference/createdocument

### PR DESCRIPTION
**Pull request by andrewjschuang**
Resolves #7815.

### <samp>🤖 Generated by Copilot at ce1f488</samp>

This pull request adds a new action component and a new app file for Affinda, and fixes the `package.json` file for the Affinda components. The new action component allows users to upload a document from a URL to Affinda for parsing. The new app file defines the common props and methods for the Affinda components, and uses the axios library to make API requests. The `package.json` file is corrected to point to the app file and remove an invalid comma.

### <samp>🤖 Generated by Copilot at ce1f488</samp>

> _`Affinda` app file_
> _Upload documents from URL_
> _Fix `package.json`_

### <samp>🤖 Generated by Copilot at ce1f488</samp>

*  Add a new action component to upload a document from a URL to Affinda ([link](https://github.com/PipedreamHQ/pipedream/pull/7881/files?diff=unified&w=0#diff-c72b849c028dd37bde678452677595313b5ad845f7e4a9849cf00647cc03b2e6R1-R109))
* Add a new app file for Affinda that provides common prop definitions and methods for the components ([link](https://github.com/PipedreamHQ/pipedream/pull/7881/files?diff=unified&w=0#diff-196a331083d64800ce9477d1c0933c36e8d2fa7d6b80174d7e5bd2ecbaa339e5R1-R80))
* Update the main field in the package.json file to point to the app file instead of the dist folder ([link](https://github.com/PipedreamHQ/pipedream/pull/7881/files?diff=unified&w=0#diff-24f234621ca5b5bcded288dab5b43490213df176591bdd720e47ebed66b72db6L5-R9))
* Remove a trailing comma in the package.json file that was causing a syntax error ([link](https://github.com/PipedreamHQ/pipedream/pull/7881/files?diff=unified&w=0#diff-24f234621ca5b5bcded288dab5b43490213df176591bdd720e47ebed66b72db6L16-R15))
* Delete unused files ([link](https://github.com/PipedreamHQ/pipedream/pull/7881/files?diff=unified&w=0#diff-e37ed4a364286c0eec83c0906c418d4eb17b2d9a24697c053d11c63b285c729d), [link](https://github.com/PipedreamHQ/pipedream/pull/7881/files?diff=unified&w=0#diff-c5342086cfe89b3c5dd5f92a61b74740af854999eccd8096a04b3e073fd0ac1d))

### CODE REVIEW

File: components/affinda/package.json [hunk 1]
```diff
@@ -1,16 +1,15 @@
 {
   "name": "@pipedream/affinda",
-  "version": "0.0.1",
+  "version": "0.0.2",
```
Comment (luancazarine) [inline components/affinda/package.json:3]:
```suggestion
  "version": "0.1.0",
```

File: components/affinda/sources/common/base.mjs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            offset,
+            limit,
+          },
+        });
+        documents.push(...results);
+        if (results.length < limit) break;
+        offset += limit;
+      }
+
+      for (const { meta: document } of documents.reverse().slice(0, 50)) {
```
Comment (luancazarine) [inline components/affinda/sources/common/base.mjs:55]: why did you sliced it by 50 if you are limiting the request to at least 300?

**Review by luancazarine — CHANGES_REQUESTED** [review]
Hi @andrewjschuang, thanks for your contribution! I just added some suggestions.

File: components/affinda/sources/common/base.mjs (hunk 2 above)
Comment (andrewjschuang) [reply components/affinda/sources/common/base.mjs:55]: Slicing the last 50 historical events, limiting the HTTP request to 300 to make less requests. Does that make sense?

**Review by luancazarine — APPROVED** [review]
LGTM!

**Pull request merged**